V9 : Port Currency Rate update

vrenaville opened the pull request:
- Remove check on base=True main currency, used company currency instead

vrenaville committed:
[IMP] Port to V9

pedrobaeza commented:
This migration should be taken as opportunity to refactor this bloated module to isolate update engine from each of the possible update sources. Maybe one or 2 of the more generals sources can be inside the main module, but not all of them.

elicoidal commented on currency_rate_update/__openerp__.py:
```diff
@@ -20,7 +20,7 @@
 {
     "name": "Currency Rate Update",
-    "version": "8.0.0.7.0",
+    "version": "9.0.0.7.0",
```
9.0.1.0.0

elicoidal commented on currency_rate_update/model/company.py:
```diff
@@ -25,15 +25,6 @@
-
     @api.one
     def button_refresh_currency(self):
         """Refresh the currencies rates !!for all companies now"""
```
Refresh the currencies rates (for all companies now).

okuryan commented:
What is status of this pull request? We need this module for our customer and would like to decide either port to v9 ourselves or use existant module

pedrobaeza commented:
As I said before, this should be refactored to have a base with base functionality and a global source, and then several submodules depending on it with the possible sources.

jgrandguillaume commented:
Hi,

We don't have the resources yet to refactor this module properly (very old one I agree). But still, lot's of people need it on v9.0. Can we accept it like this ? It is not really well designed and I agree with Pedro, but on other side it works... 

Unless someone raise a hand to refactor it, I'm giving my :+1:  to merge it once the little nits raised by Eric has been cleaned up.

Regards,

Joël

yvaucher commented:
This PR is focused to make admin ch currency update working. We don't intend to do the other as we lack of resource to do it. This could be a base for other, maybe we should raise a not ported warning for other types of currency import ?

:+1: as soon as comments from @elicoidal  are fixed

katyukha committed:
[REF] currency_rate_update: refactoring related to PR #318

Conflicts:
	currency_rate_update/services/__init__.py
	currency_rate_update/services/currency_getter.py
	currency_rate_update/services/update_service_CA_BOC.py
	currency_rate_update/services/update_service_CH_ADMIN.py
	currency_rate_update/services/update_service_ECB.py
	currency_rate_update/services/update_service_PL_NBP.py
	currency_rate_update/services/update_service_RO_BNR.py

vrenaville commented:
@yvaucher Finally I fixed reported issues

yvaucher committed:
Fix contributors, short license headers and copyrights
There was missing a bunch of people there!

yvaucher commented:
@katyukha Thanks for your work, I included it here

yvaucher committed:
Fix translated string with extra spaces

yvaucher committed:
Add a warning and a note in roadmap about Bank of canada not working

ghost commented:
Hello! What is pending in this pull request? May I help?

yvaucher commented:
@gs-clearcorp I think it was only missing reviews. We can now merge it.
